utils/gridlink_impl.c.src

```diff
@@ -45,7 +43,6 @@ void free_cellarray_DOUBLE(cellarray_DOUBLE *lattice, const int64_t totncells)
                 free(lattice[i].x);
                 free(lattice[i].y);
                 free(lattice[i].z);
-                free(lattice[i].original_index);
```

Comment:
May be this deletion will become clearer later, but seems odd at first sight. We will need the `original_index` for `copy_particles == 0`

Reply:
This is only deleting the `original_index` for `copy_particles == 1`.  `copy_particles == 0` never owns its memory, so it never enters this branch of the `if(lattice[0].owns_memory == 0)` test.

Which reminds me: was there any reason we needed to keep `original_index` when `copy_particles == 1`?  I think the old code was keeping it regardless of `copy_particles == 1/0`.

Reply:
That was there a possible feature where the original indices are "yielded" for each pair

Reply:
Did we want to preserve that possibility, or is it okay to remove the original indices when copying the particles?

Reply:
Let's leave that in for the moment - less to think about :)

Reply:
Hmm - does this mean that `original_index` is only created when `copy_particles==False`? Otherwise, there will be a memory leak

Reply:
I am not sure that `original_index` contained anything other than `NULL` when `copy_particles==True` but will you please confirm.